content/structure.md

```diff
@@ -0,0 +1,390 @@
+# Application structure and code style
+
+After reading this article, you'll know:
+
+1. How a Meteor application compares to other types of applications in terms of file structure.
+2. How to organize your application both for small and larger applications.
+3. How to format your code and name the parts of your application in consistent and maintainable ways.
+
+<h2 id="meteor-structure">Meteor Application Structure</h2>
+
+Meteor is a *full-stack* framework for building applications; this means Meteor applications differ from most applications in that they include code that both runs on the client, code that runs on the server, and code that runs in *both places*. Meteor applications enable you to run JavaScript code easily and consistenly in both client and server environments, and some conventions on application structure makes it easier to understand what code runs where.
+
+<h3 id="es2015-imports">ES2015 Imports</h3>
+
+As of version 1.3, Meteor ships with full support for ES2015 packages, both on the client and on the server. In particular it is simple and straightforward to include NPM package code in either environment, and to write your code in a modular way, avoiding the introduction of global symbols and "action at a distance". 
+
+However, this is a new feature in Meteor so you will find a lot of code online that uses and older, more centralized system built around packages and app declaring global symbols to be consumed within the app. Also, for backwards compatibility reasons, "pure" modular code must be placed in the `imports/` directory in your application. We expect this to change in an future release.
+
+<h2 id="javascript-structure">JavaScript file structure</h2>
+
+To harness the module system and ensure that our code only runs when we specify that it should, all code for libraries, components and APIs should be placed in the `imports/` directory. This will mean that the Meteor build system will only bundle and include that code if it is `import`-ed from another file.
+
+Meteor will eagerly load any files outside of `imports/` in the application, but to be explicit, it's best to create a `main/client` and `main/server` folder to define explicit entry points for the running code of your application. 
+
+In the Todos example app, the `main/client/routes.js` indirectly file imports *all* other code that is required on the client, and forms the single entry point for the client application:
```

Comment:
I'm not sure really. What could also make sense to me would be to have `client/main.js` that imports the routes, because:

 1. You might want to import the routes on the server
 2. Your router file could be really huge and want to be split over multiple files.

Also there can be other client init stuff like:
 1. Configuring things like intercom / GA / segment / etc.
 2. Configuring accounts related stuff.

I'm not convinced it's not better to just have 2 simple entrypoints `client/main.js` and `server/main.js`. Although this means you probably need like `import/initializers` for random config type stuff. I'd be OK with that though.